corehq/apps/data_interfaces/forms.py

```diff
@@ -270,9 +307,12 @@ def __init__(self, *args, **kwargs):
             ),
         )
 
+    def _valid_boundary(self, value):
+        return self.enhancements_enabled or (isinstance(value, int) and value > 30)
+
     def clean_server_modified_boundary(self):
         value = self.cleaned_data.get('server_modified_boundary')
-        if not isinstance(value, int) or value < 30:
+        if not self._valid_boundary(value):
             raise ValidationError(_("Please enter a whole number greater than "
                                     "or equal to 30."))
         return value
```

Comment:
we should make this return None if we're not filtering on server modified boundary, so that it gets set to None in the rule